dissect/target/plugins/os/unix/trash.py

```diff
@@ -0,0 +1,124 @@
+from typing import Iterator
+
+from dissect.target.exceptions import UnsupportedPluginError
+from dissect.target.helpers import configutil
+from dissect.target.helpers.descriptor_extensions import UserRecordDescriptorExtension
+from dissect.target.helpers.fsutil import TargetPath
+from dissect.target.helpers.record import create_extended_descriptor
+from dissect.target.plugin import Plugin, alias, export
+from dissect.target.plugins.general.users import UserDetails
+from dissect.target.target import Target
+
+TrashRecord = create_extended_descriptor([UserRecordDescriptorExtension])(
+    "linux/filesystem/recyclebin",
+    [
+        ("datetime", "ts"),
+        ("path", "path"),
+        ("filesize", "filesize"),
+        ("path", "deleted_path"),
+        ("path", "source"),
+    ],
+)
+
+
+class GnomeTrashPlugin(Plugin):
+    """Linux GNOME Trash plugin.
+
+    Recovers deleted files and artifacts from ``$HOME/.local/share/Trash``.
+    Probably also works with other desktop interfaces as long as they follow
+    the Trash specification from FreeDesktop.
+
+    Currently does not parse media trash locations such as ``/media/$Label/.Trash-1000/*``.
+
+    Resources:
+        - https://specifications.freedesktop.org/trash-spec/latest/
+        - https://github.com/GNOME/glib/blob/main/gio/glocalfile.c
+        - https://specifications.freedesktop.org/basedir-spec/latest/
+    """
+
+    PATHS = [
+        # Default $XDG_DATA_HOME/Trash
+        ".local/share/Trash",
+    ]
+
+    def __init__(self, target: Target):
+        super().__init__(target)
+        self.trashes = list(self.garbage_collector())
+
+    def garbage_collector(self) -> Iterator[tuple[UserDetails, TargetPath]]:
+        for user_details in self.target.user_details.all_with_home():
+            for trash_path in self.PATHS:
+                if (path := user_details.home_path.joinpath(trash_path)).exists():
+                    yield user_details, path
+
+    def check_compatible(self) -> None:
+        if not self.trashes:
+            raise UnsupportedPluginError("No Trash folder(s) found")
+
+    @export(record=TrashRecord)
+    @alias(name="recyclebin")
+    def trash(self) -> Iterator[TrashRecord]:
```

Comment:
Please add some record field documentation. Could you also document some of the caveats here as well? For example the timestamp technicality for expunged files.

Reply:
Something along the lines of what is shown below. This will be used in the plugin documentation and will show up when you use `target-query -f trash -h`.

https://github.com/fox-it/dissect.target/blob/main/dissect/target/plugins/apps/texteditor/windowsnotepad.py#L307-#L316

Reply:
Yes, that would be nice!